cirq/ops/two_qubit_diagonal_gate.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2020 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Creates the Gate instrance for a two qubit diagonal gate.
+
+The Gate is used to create a 4x4 matrix with the diagonal elements
+passed as a list.
+"""
+
+from typing import Any, Tuple, List, TYPE_CHECKING
+import numpy as np
+import sympy
+
+from cirq import protocols, value
+from cirq._compat import proper_repr
+from cirq.ops import gate_features
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    import cirq
+
+
+@value.value_equality()
+class TwoQubitDiagonalGate(gate_features.TwoQubitGate):
+    """A gate given by a diagonal 4x4 matrix."""
```

Comment:
Use latex notation for multiplication sign.

Reply:
Has been addressed.